addons/hearing/functions/fnc_addEarPlugs.sqf

```diff
@@ -15,9 +15,12 @@
  */
 #include "script_component.hpp"
 
-private ["_unit", "_launcher"];
+PARAMS_1(_unit);
 
-_unit = _this select 0;
+// Exit if soldier has earplugs already in (persistence scenarios)
+if (FUNC(hasEarPlugsIn)) exitWith {};
```

Comment:
"Error if: Type code, expected Bool"
probably should be:
`if ([_unit] call FUNC(hasEarPlugsIn)) exitWith {};`